posts/2024-07-08_unix_vs_sas_date.../unix_vs_sas_datetime.qmd

```diff
@@ -0,0 +1,100 @@
+
+In SAS, the epoch is defined as midnight on January 1, 1960.
+(01JAN1960:00:00:00).
+
+R uses the Unix epoch which begins epoch begins at midnight on January 1, 1970 (01JAN1970:00:00:00)
+
+## Date Differences Between R and SAS
+
+Another important aspect to consider is the difference in data types for numeric dates in R and SAS.
+
+-   **SAS:** SAS only supports numeric and character data types.
+    Therefore, numeric dates in SAS need a format applied to be human-readable.
+
+-   **R:** In contrast, R has specific data types for dates, which allow numeric dates to be rendered with a date format directly in data frames.
+    The data types in R are: "Date" for dates, "hms" or "difftime" for time, and "POSIXct" or "POSIXt" for datetime.
+
+These differences can lead to discrepancies if not properly accounted for during data conversion, potentially resulting in a date shift of 10 years when transferring data between SAS and R.
+
+## Considerations for Conversion
+
+To illustrate how date conversions between SAS and R can be managed, let's look at the [`haven`](https://haven.tidyverse.org/) package in R.
+This package facilitates the conversion of data between SAS and R.
+
+-   **From SAS to R:** Numeric variables with dates formats are identified as dates and converted to POSIX date(time) formats.
+
+    **Detection of date, time, datetime variable from SAS by haven package**
+
+    ![](haven_sas_to_R.png)
+
+-   **From R to SAS:** POSIX date variables are converted to SAS numeric dates.
+    By default, a format is added to these dates to make them human-readable.
+    This format can be customized by changing the attribute or by using the [`xportr`](https://atorus-research.github.io/xportr/) package to pull the format from metadata and apply it to the corresponding variable.
+
+    **Detection of date, time, datetime variable from R by haven package**
+
+    ![](haven_R_to_SAS.PNG)
+
+## Dealing with Date Interoperability in Dataset-JSON
+
+[Dataset-JSON v1.1](https://wiki.cdisc.org/display/DSJSON1DOT1/Specification) could potentially be used in future data exchanges between SAS and R.
+Given the differing approaches to handling dates across programming languages, the challenge lies in managing dates within a programming language agnostic data exchange format.
+
+**What is Dataset-JSON?** Dataset-JSON is a dataset exchange standard that uses JSON.
+It is designed to meet regulatory submission requirements and other dataset exchange scenarios.
+It has the potential to replace XPT as the default format for clinical and device data submissions to regulatory authorities.
+
+To address the interoperability issue across programming languages, Dataset-JSON stores numeric date variables as strings in ISO 8601 format.
+Additionally, a `targetdatatype` attribute is included to differentiate between character and numeric dates within the dataset.
+This approach ensures that conversion tools receive sufficient information to handle dates correctly.
```

Comment:
Can we give a shout out to this R package as well! https://atorus-research.github.io/datasetjson/